test/CAPI/firrtl.c

```diff
@@ -79,10 +79,42 @@ void testValueFoldFlow(MlirContext ctx) {
          FIRRTL_VALUE_FLOW_SINK);
 }
 
+void testImportAnnotations(MlirContext ctx) {
+  // clang-format off
+  const char *testFIR =
+    "firrtl.circuit \"AnnoTest\" {\n"
+    "  firrtl.module @AnnoTest(in %in: !firrtl.uint<32>) {}\n"
+    "}\n";
+  // clang-format on
+  MlirModule module =
+      mlirModuleCreateParse(ctx, mlirStringRefCreateFromCString(testFIR));
+  MlirBlock mlirModule = mlirModuleGetBody(module);
+  MlirOperation firCircuit = mlirBlockGetFirstOperation(mlirModule);
+
+  const char *rawAnnotationsJSON = "[{\
+    \"class\":\"firrtl.transforms.DontTouchAnnotation\",\
+    \"target\":\"~AnnoTest|AnnoTest>in\"\
+  }]";
+  MlirAttribute rawAnnotationsAttr;
+  assert(firrtlImportAnnotationsFromJSONRaw(
```

Comment:
FYI @SpriteOvO , some builds may disable asserts, and we do test with asserts disabled in nightly CI. I missed this in review: this call shouldn't be wrapped in assert since it does meaningful work needed for the test to function. I changed it here: https://github.com/llvm/circt/commit/ed19a32f865acf1055635310f143f8959dcec121

Reply:
Oh! Once again, I forgot about this. 🤦 Thanks for your fix.

Reply:
No problem, I will try to keep a better look out for it in review as well.